packages/compose/src/hooks/use-focus-return/index.js

```diff
@@ -33,53 +33,44 @@ import useCallbackRef from '../use-callback-ref';
  * ```
  */
 function useFocusReturn( onFocusReturn ) {
+	const ref = useRef();
 	const focusedBeforeMount = useRef();
-	const isFocused = useRef( false );
 	const onFocusReturnRef = useRef( onFocusReturn );
 	useEffect( () => {
 		onFocusReturnRef.current = onFocusReturn;
-	}, [] );
-
-	const ref = useCallbackRef( ( newNode ) => {
-		const updateLastFocusedRef = ( { target } ) => {
-			isFocused.current = newNode && newNode.contains( target );
-		};
+	}, [ onFocusReturn ] );
 
-		// Unmounting the reference
-		if ( ! newNode && focusedBeforeMount.current ) {
-			if ( newNode?.ownerDocument ) {
-				newNode.ownerDocument.removeEventListener(
```

Comment:
This PR also fixes a memory leak. This line will never execute and even if it did, `updateLastFocusedRef` will be a different function because it is different every time the ref callback is called.